### PR TITLE
fix: jsonSerialize return type array

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
Missing return type for jsonSerialise cause a massive amount of warnings in logs.

### CODE REVIEW

**Comment by kstelmaczonek** [comment]
Its better now :) But still its 30+ files change so there's could be collateral damage done can we test all those changes?

File: models/classes/taskQueue/TaskLog/Entity/TaskLogEntity.php [hunk 1]
```diff
@@ -311,12 +313,19 @@ public function toArray()
 
         if ($this->createdAt instanceof \DateTime) {
             $rs['createdAt'] = $this->createdAt->getTimestamp();
-            $rs['createdAtElapsed'] = (new \DateTime('now', new \DateTimeZone('UTC')))->getTimestamp() - $this->createdAt->getTimestamp();
+            $rs['createdAtElapsed'] = (new \DateTime(
```
Comment (Karol-Stelmaczonek) [inline models/classes/taskQueue/TaskLog/Entity/TaskLogEntity.php:316]: This formatted block doesn't look good. Below is almost the same and look normal :)

**Comment by Karol-Stelmaczonek** [comment]
Piece of good work @bartlomiejmarszal , but from experience I know something can blow up on those return types. Strange edge cases with nulls especially, also we need to check it with PHP8 in mind. Maybe we are covered IDK.

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by andreluizmachado — CHANGES_REQUESTED** [review]
Hey @bartlomiejmarszal,
This PR seems to contain breaking changes for me so I'd leave commits as bellow to follow semantic versioning:
```
fix!: ....
or
fix: xxxx
BREAKING CHANGE: yyy
```

File: models/classes/datatable/implementation/AbstractDatatablePayload.php [hunk 2]
```diff
@@ -286,10 +286,7 @@ protected function fetchPropertyValues($payload)
         return $payload;
     }
 
-    /**
-     * @return array
-     */
-    public function jsonSerialize()
+    public function jsonSerialize(): array
     {
         return $this->getPayload();
```
[the remark this answers is not in the document]
Comment (bartlomiejmarszal) [reply models/classes/datatable/implementation/AbstractDatatablePayload.php:285]: I will put the correct return types for all related methods.

File: models/classes/taskQueue/Task/CallbackTask.php [hunk 3]
```diff
@@ -98,7 +98,7 @@ public function isEnqueued()
     /**
      * @return array
      */
-    public function jsonSerialize()
+    public function jsonSerialize(): array
     {
         $callableClassOrArray = $this->getCallable();
```
[the remark this answers is not in the document]
Comment (bartlomiejmarszal) [reply models/classes/taskQueue/Task/CallbackTask.php:103]: But it is not callable that is returned. 
```
return parent::jsonSerialize();
```
And it is always array

File: models/classes/service/class.Parameter.php [hunk 4]
```diff
@@ -56,12 +56,13 @@ public function getDefinition()
...
         return $this->definition;
     }
+
     /**
      * @return core_kernel_classes_Resource
      */
     abstract public function toOntology(Ontology $model);
 
-    abstract public function jsonSerialize();
+    abstract public function jsonSerialize(): array;
```
[the remark this answers is not in the document]
Comment (bartlomiejmarszal) [reply models/classes/service/class.Parameter.php]: I reverted this change

File: models/classes/taskQueue/Task/CallbackTask.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (andreluizmachado) [reply models/classes/taskQueue/Task/CallbackTask.php:103]: can you please add return type here?
```
    /**
     * @return callable|string|object
     */
    public function getCallable()
    ```

[the remark this answers is not in the document]
Comment (bartlomiejmarszal) [reply models/classes/taskQueue/Task/CallbackTask.php:103]: There is many other return types missing. I want to fix only those that are causing errors this time.
